parser/match_operations.go

```diff
@@ -437,16 +437,33 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
+
+		match0Ops := matches[amountMatch[0]].Operations
+		match1Ops := matches[amountMatch[1]].Operations
+
+		if len(match0Ops) == 0 || len(match1Ops) == 0 {
+			return errors.New("opposite amounts comparison error: no matched operations")
```

Comment:
On further thought, I like the idea of breaking this into 2 if statements so we can surface a clearer error message.

nit: description `amountMatch[0]` does not have enough operations